# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1378359466 14400
#      Thu Sep 05 01:37:46 2013 -0400
# Node ID bc018154e46a0ce4547b71e4c5c0d267e6236b96
# Parent  991c7c812e3892375a3b9efd44403970a11fd3a7
build: Support optional Automake silent rules

* configure.ac: Call AM_SILENT_RULES to provide --enable-silent-rules.
* libinterp/Makefile.am (.yy.cc): Add Automake silent rule prefix.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -38,16 +38,20 @@ dnl        AC_REVISION field whenever co
 dnl AC_REVISION($Revision: 1.603 $)
 AC_CONFIG_SRCDIR([libinterp/octave.cc])
 AC_CONFIG_HEADERS([config.h:config.in.h])
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_MACRO_DIR([m4])
 
 AM_INIT_AUTOMAKE([1.11 tar-ustar])
 
+## Add the option to enable silent rules, available since Automake 1.11
+## and included by default starting with Automake 1.13.
+AM_SILENT_RULES
+
 OCTAVE_CANONICAL_HOST
 
 AC_DEFINE(OCTAVE_SOURCE, 1, [Define to 1 if this is Octave.])
   
 AC_USE_SYSTEM_EXTENSIONS
 
 ### Make configure args available for other uses.
 
diff --git a/libinterp/Makefile.am b/libinterp/Makefile.am
--- a/libinterp/Makefile.am
+++ b/libinterp/Makefile.am
@@ -251,17 +251,17 @@ TST_FILES := $(addsuffix -tst,$(TST_FILE
 libinterptestsdir := $(octtestsdir)/libinterp
 
 nobase_libinterptests_DATA = $(TST_FILES)
 
 ## Override Automake's rule that forces a .hh extension on us even
 ## though we don't want it.  It would be super awesome if automake
 ## would allow users to choose the header file extension.
 .yy.cc:
-	$(am__skipyacc) $(SHELL) $(YLWRAP) $< y.tab.c $@ y.tab.h $*.h y.output $*.output -- $(YACCCOMPILE)
+	$(AM_V_YACC)$(am__skipyacc) $(SHELL) $(YLWRAP) $< y.tab.c $@ y.tab.h $*.h y.output $*.output -- $(YACCCOMPILE)
 
 ## Special rules:
 ## Mostly for sources which must be built before rest of compilation.
 
 ## oct-conf.h must depend on Makefile.
 ## Calling configure may change default/config values.
 ## However, calling configure will also regenerate the Makefiles from
 ## Makefile.am and trigger the rules below.
